format-parquet/src/test/java/io/cdap/plugin/format/parquet/input/ParquetInputFormatProviderTest.java

```diff
@@ -74,4 +76,43 @@ public void testSchemaDetection() throws IOException {
     Schema schema = schemaDetector.detectSchema(parquetFile.getAbsolutePath(), formatContext, Collections.emptyMap());
     Assert.assertEquals(expectedSchema, schema);
   }
+
+  @Test
+  public void testPathFieldSchemaDetection() throws IOException {
+    boolean existingPathFieldValue = false;
+    try {
+      Configuration hConf = new Configuration();
+      File parquetFile = new File(TMP_FOLDER.newFolder(), "test.parquet");
+      Path parquetPath = new Path(parquetFile.toURI());
+
+      Schema expectedSchema = Schema.recordOf("x",
+        Schema.Field.of("id", Schema.of(Schema.Type.INT)),
+        Schema.Field.of("name", Schema.of(Schema.Type.STRING)),
+        Schema.Field.of("score", Schema.of(Schema.Type.DOUBLE)),
+        Schema.Field.of("pathField", Schema.of(Schema.Type.STRING)));
+      org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(expectedSchema.toString());
+      GenericRecord record = new GenericRecordBuilder(avroSchema)
+        .set("id", 0)
+        .set("name", "alice")
+        .set("score", 1.0d)
+        .set("pathField", "value")
+        .build();
+
+      ParquetWriter<GenericRecord> parquetWriter =
+        AvroParquetWriter.<GenericRecord>builder(HadoopOutputFile.fromPath(parquetPath, hConf))
+          .withSchema(avroSchema)
+          .build();
+      parquetWriter.write(record);
+      parquetWriter.close();
+
+      ParquetInputFormatProvider.Conf conf = new ParquetInputFormatProvider.Conf("pathField");
+      ParquetInputFormatProvider formatProvider = new ParquetInputFormatProvider(conf);
+      FormatContext formatContext = new FormatContext(new MockFailureCollector(), null);
+      SchemaDetector schemaDetector = new SchemaDetector(formatProvider);
+      Schema schema = schemaDetector.detectSchema(parquetFile.getAbsolutePath(), formatContext, Collections.emptyMap());
+    } catch (Exception e) {
```

Comment:
If the test is expected to throw an exception, the test should be annotated with the exception class

```
@Test(expected = SomeExceptionClass.class)
```

Reply:
Or at least you should verify the exception thrown is the class you expect it to be.

Reply:
In general, please follow Fernando's suggestion when a test is expected to throw a specific type of exception.

For this particular test case, this should be checking that the proper failure is being added to the FailureCollector instead of expecting an exception

Reply:
Noted.Added Failure Collector